### DISCUSSION TITLE
PMD 7.0 RuleChain deprecated

### DISCUSSION THREAD

**Discussion by MichaelPStimpson** [pmd/pmd#3636] [Q&A] [answered]
Hello, I'm writing some custom rules and while digging through the repo and documentation I found several instances of "RuleChain" (In the apex rules) being deprecated.  

Is RuleChain being deprecated in favor of something else?
(If I missed some documentation feel free to point me back to the docs 😅)

**Comment by MichaelPStimpson** [comment, marked as answer]
I checked the 7.x branch and see the rewrite using RuleTargetSelector... 😅

**Reply by oowekyala** [reply]
Yeah, only using the RuleChain class directly is deprecated. The rulechain as a performance feature remains, with a different (and mostly hidden) implementation
